scripts/us_census/pep/pep_by_asrh/preprocess.py

```diff
@@ -0,0 +1,1072 @@
+    origin_mapper = {1: "NotHispanicOrLatino", 2: "HispanicOrLatino"}
+    race_mapper = {
+        0: "empty",
+        1: "WhiteAlone",
+        2: "BlackOrAfricanAmericanAlone",
+        3: "AmericanIndianAndAlaskaNativeAlone",
+        4: "AsianAlone",
+        5: "NativeHawaiianAndOtherPacificIslanderAlone",
+        6: "TwoOrMoreRaces"
+    }
+    # Deriving New Columns and actual values for below numbers are
+    # represented in above dictonaries
+    for origin in [1, 2]:
+        derived_cols_data_df = pd.concat([
+            derived_cols_data_df,
+            data_df[(data_df["ORIGIN"] == origin) & (data_df["SEX"] == 0) &
+                    (data_df["RACE"] == 0) & (data_df["AGEGRP"] != 0)]
+        ],
+                                         ignore_index=True)
+        for sex in [0, 1, 2]:
+            if sex == 0:
+                for race in [1, 2, 3, 4, 5, 6]:
+                    derived_cols_data_df = pd.concat([
+                        derived_cols_data_df,
+                        data_df[(data_df["ORIGIN"] == origin) &
+                                (data_df["SEX"] == 0) &
+                                (data_df["RACE"] == race) &
+                                (data_df["AGEGRP"] != 0)]
+                    ],
+                                                     ignore_index=True)
+            else:
+                derived_cols_data_df = pd.concat([
+                    derived_cols_data_df,
+                    data_df[(data_df["ORIGIN"] == origin) &
+                            (data_df["SEX"] == sex) & (data_df["RACE"] == 0) &
+                            (data_df["AGEGRP"] != 0)]
+                ],
+                                                 ignore_index=True)
+    data_df = data_df[(data_df["SEX"] != 0) & (data_df["RACE"] != 0) &
+                      (data_df["ORIGIN"] != 0) &
+                      (data_df["AGEGRP"] != 0)].reset_index(drop=True)
+    data_df = pd.concat([data_df, derived_cols_data_df], ignore_index=True)
+    data_df["RACE"] = data_df["RACE"].map(race_mapper)
+    data_df["ORIGIN"] = data_df["ORIGIN"].map(origin_mapper)
+    data_df["SEX"] = data_df["SEX"].map(gender_mapper)
+    data_df[
+        "SV"] = data_df["SEX"] + "_" + data_df["ORIGIN"] + "_" + data_df["RACE"]
+    data_df["SV"] = data_df["SV"].str.replace("NotHispanicOrLatino_WhiteAlone",
+                                              "WhiteAloneNotHispanicOrLatino")
+    data_df["SV"] = data_df["SV"].str.replace("empty_",
+                                              "").str.replace("_empty", "")
+    data_df["Location"] = "geoId/" + data_df["STATE"].astype("str").str.pad(
+        width=2, side="left", fillchar="0")
+    cols = data_df.columns.to_list()
+    # Creating Age Groups
+    data_df["Age"] = data_df["AGEGRP"].apply(_get_age_grp)
+    data_df["SV"] = data_df.apply(lambda row: _create_sv(row.SV, row.Age),
+                                  axis=1)
+    cols = ["Location", "SV"
```

Comment:
This isn't great for readability. You are using 'cols' in a conditional iterator which produces a new temporary list and then append to [Location, SV] list and then overwrite the 'cols' list. And then two lines below you are overwriting 'cols' again. I suggest either using helper functions or creating useful local variables for clarity. This is also something I recommend fixing everywhere, as needed.